### PR TITLE
HIP-584: Fix transfer precompile alias conditions

### PR DESCRIPTION
**Pull request by stoyanov-st**
**Description**:
Fix condition that was unnecessarily creating aliases for recipients of crypto transfer functions.
Provided integration tests for transfers with aliases.

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServicePrecompileTest.java [hunk 1]
```diff
@@ -344,18 +344,36 @@ enum SupportedContractModificationFunctions {
...
+        }),
+        CRYPTO_TRANSFER_TOKENS(
+                "cryptoTransferExternal", new Object[] {TREASURY_TOKEN_ADDRESS, SENDER_ADDRESS, SPENDER_ADDRESS, 5L}),
+        CRYPTO_TRANSFER_TOKENS_WITH_ALIAS(
+                "cryptoTransferExternal", new Object[] {TREASURY_TOKEN_ADDRESS, SENDER_ALIAS, SPENDER_ALIAS, 5L}),
         TRANSFER_NFT_TOKENS("transferNFTsExternal", new Object[] {
             NFT_TRANSFER_ADDRESS, new Address[] {OWNER_ADDRESS}, new Address[] {SPENDER_ADDRESS}, new long[] {1}
         }),
+        TRANSFER_NFT_TOKENS_WITH_ALIAS("transferNFTsExternal", new Object[] {
+            NFT_TRANSFER_ADDRESS, new Address[] {OWNER_ADDRESS}, new Address[] {SPENDER_ALIAS}, new long[] {1}
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServicePrecompileTest.java:363]: Probable we can create `OWNER_ALIAS` to be compliant with the other test scenarios.

Comment (stoyanov-st) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServicePrecompileTest.java:462]: We can provide it with the refactoring in the next package.

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/codec/DecodingFacade.java [hunk 2]
```diff
@@ -147,7 +147,7 @@ public static List<FungibleTokenTransfer> bindFungibleTransfersFrom(
         for (final var transfer : abiTransfers) {
             var accountID = convertLeftPaddedAddressToAccountId(transfer.get(0), aliasResolver);
             final long amount = transfer.get(1);
-            if (amount > 0 && !accountID.hasAlias()) {
+            if (accountID.hasAlias()) {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/codec/DecodingFacade.java:150]: In services we have the following logic:

```
final long amount = transfer.get(1);
if (amount > 0 && !exists.test(accountID) && !accountID.hasAlias()) {
   accountID = generateAccountIDWithAliasCalculatedFrom(accountID);
}
```
I think we should follow the same checks and introduce an `exists` lambda function, that checks if a given account is present in the `Store`.

Comment (stoyanov-st) [reply hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/codec/DecodingFacade.java]: 'exists' method added and the condition is set accordingly.

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/codec/TransferParams.java [hunk 3]
```diff
@@ -16,6 +16,7 @@
 
 package com.hedera.services.store.contracts.precompile.codec;
 
-import org.hyperledger.besu.datatypes.Address;
+import com.hederahashgraph.api.proto.java.AccountID;
+import java.util.function.Predicate;
 
-public record TransferParams(int functionId, Address sernderAddress) implements BodyParams {}
+public record TransferParams(int functionId, Predicate<AccountID> setExistsLambda) implements BodyParams {}
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/codec/TransferParams.java:22]: Rename `setExistsLambda` -> `exists`

Comment (stoyanov-st) [reply hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/codec/TransferParams.java]: Done

**Review by IvanKavaldzhiev — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**